docs/dev-guides/PERSONAL-ENVIRONMENTS.md

```diff
@@ -0,0 +1,523 @@
+# Personal Test Environments
+Personallly provisioned test environments, and the steps to create and manage them, are described here.
+## Background
+### Use Case
+Personal Kubernetes environments allow realistic testing of deployed services without interfering
+with shared environments. This enables faster development cycles, as code can be tested without
+waiting for a release, and without risk of breaking colleagues' environments.
+
+### Where they Live
+A personal environment is contained by a K8s [namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/)
+named for a developer, sharing a larger cluster in the GCP
+project [terra-kernel-k8s](https://console.cloud.google.com/home/dashboard?project=terra-kernel-k8s)
+called 
+[terra-integration](https://console.cloud.google.com/kubernetes/clusters/details/us-central1-a/terra-integration?project=terra-kernel-k8s&tab=details&persistent_volumes_tablesize=50&storage_class_tablesize=50&nodes_tablesize=50&node_pool_tablesize=10).
+Deployments target this namespace and allow `kubectl` commands to be narrowly focused. The deployment
+consists of a single
+[pod](https://kubernetes.io/docs/concepts/workloads/pods/) and [service](https://kubernetes.io/docs/concepts/services-networking/service/)
+in the case of Workspace Manager, though the system is more flexible than that.
+
+### Helpful Background Reading
+#### Kubernetes
+[The Kubernetes Book]() is very accessible for beginners (like myself), and uses lots of images and
+repetition to bring the main points home. Additionally, the exercises on the main Kubernetes site are
+well-paced, insightful, and a realistic facsimile of a k8s cluster and the process of interacting
+with it. The [kubectl Cheat Sheet](https://kubernetes.io/docs/reference/kubectl/cheatsheet/) has lots
+of useful goodies.
+
```

Comment:
Not sure where this belongs, but early on @wchamber explained to me the difference in how we use terraform and helm which was really useful. Basically, terraform brings up and configures all resources surrounding the core service (storage, iam, etc), while helm basically just provides environment variables to the running service. Also worth early on stating that we have this hierarchy in both cases where there is a generic thing in one repo and then a specific layer you need to add on in another, private repo.

Reply:
Whoa, so helm is almost like Zookeeper in that model. I thought it also deployed the services pods via their helm charts.

Reply:
Helm does more than providing environment variables to the running service - it's our template for all configuration/specification within k8s. The line between terraform & helm is what each is more targeted to do: terraform configure cloud resources & helm for everything in k8s land.